### PR TITLE
[Back-to-top] Add Back-to-top button for all pages

### PR DESCRIPTION
**Pull request by BlueDaisy061**
### Description
- Implement UI for Back-to-top button
- Add Back-to-top to Layout

### Screenshot
image

### CODE REVIEW

File: src/components/BackToTop/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      const scrollToTop = () =>{
+        window.scrollTo({
+          top: 0, 
+          behavior: 'smooth'
+        });
+      };
+      
+      window.addEventListener('scroll', toggleVisible);
+  return (
+    <button className={styles.backToTop} style={{display: visible ? 'inline' : 'none'}} >
```
Comment (tainguy22112000) [inline src/components/BackToTop/index.tsx]: you can use classnames here, for example
className={ classNames(styles.backToTop, visible && styles.visible) }

File: src/components/BackToTop/index.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react'
+import { FaChevronUp } from 'react-icons/fa'
+import styles from './styles.module.scss'
+
+export default function BackToTop() {
+    const [visible, setVisible] = useState<boolean>(false);
+
+    const toggleVisible = () => {
+        const scrolled = document.documentElement.scrollTop;
+        if (scrolled > 300){
```
Comment (tainguy22112000) [inline src/components/BackToTop/index.tsx]: declare a constant on the top of the component.

Comment (BlueDaisy061) [reply src/components/BackToTop/index.tsx]: sorry can I ask why we need to do that?

File: src/components/BackToTop/index.tsx (hunk 1 above)
Comment (BlueDaisy061) [reply src/components/BackToTop/index.tsx]: cool! updated [465d3ce](https://github.com/tainguy22112000/DaisyAgency/pull/6/commits/465d3cea42eb64fb5ac8852c98f2d12882f4cbaf)

File: src/components/BackToTop/index.tsx (hunk 2 above)
Comment (tainguy22112000) [reply src/components/BackToTop/index.tsx]: It's related to cleaning code (avoid magic numbers). The best way to fix magic numbers is to move them into their own constant variables. This way the variable name of the magic number can describe exactly what the variable does. In this case, you can declare a constant : 
const SCROLL_THESHOLD = 300

**Review by tainguy22112000 — APPROVED** [review]

**Review by tainguy22112000 — APPROVED** [review]

**Pull request merged**